### PR TITLE
feat(modelarts): add resource to operate devserver

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add new resource to start or stop DevService.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource to operate DevService.
2. add related document and accptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
 ./scripts/coverage.sh -o modelarts -f TestAccDevServer_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/modelarts" -v -coverprofile="./huaweicloud/services/acceptance/modelarts/modelarts_coverage.cov" -coverpkg="./huaweicloud/services/modelarts" -run TestAccDevServer_basic -timeout 360m -parallel 10
=== RUN   TestAccDevServer_basic
=== PAUSE TestAccDevServer_basic
=== CONT  TestAccDevServer_basic
--- PASS: TestAccDevServer_basic (433.24s)
PASS
coverage: 10.1% of statements in ./huaweicloud/services/modelarts
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts 433.329s        coverage: 10.1% of statements in ./huaweicloud/services/modelarts
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

**Comment by wuzhuanhong** [comment]
./scripts/coverage.sh -o modelarts -f TestAccDevServer_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/modelarts" -v -coverprofile="./huaweicloud/services/acceptance/modelarts/modelarts_coverage.cov" -coverpkg="./huaweicloud/services/modelarts" -run TestAccDevServer_basic -timeout 360m -parallel 10
=== RUN   TestAccDevServer_basic
=== PAUSE TestAccDevServer_basic
=== CONT  TestAccDevServer_basic
--- PASS: TestAccDevServer_basic (513.42s)
PASS
coverage: 10.1% of statements in ./huaweicloud/services/modelarts
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts 513.504s        coverage: 10.1% of statements in ./huaweicloud/services/modelarts

**Comment by wuzhuanhong** [comment]
./scripts/coverage.sh -o modelarts -f TestAccDevServer_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/modelarts" -v -coverprofile="./huaweicloud/services/acceptance/modelarts/modelarts_coverage.cov" -coverpkg="./huaweicloud/services/modelarts" -run TestAccDevServer_basic -timeout 360m -parallel 10
=== RUN TestAccDevServer_basic
=== PAUSE TestAccDevServer_basic
=== CONT TestAccDevServer_basic
--- PASS: TestAccDevServer_basic (513.42s)
PASS
coverage: 10.1% of statements in ./huaweicloud/services/modelarts
ok github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts 513.504s coverage: 10.1% of statements in ./huaweicloud/services/modelarts

File: huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_devserver_test.go [hunk 1]
```diff
@@ -88,7 +106,7 @@ func TestAccDevServer_basic(t *testing.T) {
 	})
 }
 
-func testDevServer_basic(name, password string, autoRenew bool) string {
+func testDevServer_basic_step1(name, password string, autoRenew bool) string {
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_devserver_test.go]: rename it as the `testAccDevServer_basic` and using these references as follows:
```
// Rename the argument input 'autoRenew' as the 'isAutoRenew', and put it ahead of the other string arguments.
func testAccDevServer_basic(isAutoRenew bool, name, password string) string {
  return testAccDevServer_base(isAutoRenew, name, password)
}
```

File: huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_devserver_test.go [hunk 2]
```diff
@@ -114,3 +132,51 @@ resource "huaweicloud_modelarts_devserver" "test" {
 `, common.TestBaseNetwork(name), name,
 		acceptance.HW_MODELARTS_DEVSERVER_FLAVOR, acceptance.HW_MODELARTS_DEVSERVER_IMAGE_ID, password, autoRenew)
 }
+
+func testDevServer_basic_step2(name, password string, autoRenew bool) string {
+	return testDevServer_basic_step1(name, password, autoRenew)
+}
+
+func testDevServer_basic_step3(name, password string, autoRenew bool) string {
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_devserver_test.go]:
```
testAccDevServer_doAction(isAutoRenew bool, name, password string, actionType string, doRetryAction bool) string {
    return fmt.Sprintf(`
%[1]s

resource "huaweicloud_modelarts_devserver_action" "test" {
  devserver_id = huaweicloud_modelarts_devserver.test.id
  action       = "%[2]s"
}

variable "is_retry_devserver_action" {
  type    = bool
  default = %[3]s
}

resource "huaweicloud_modelarts_devserver_action" "retry" {
  count = var.is_retry_devserver_action ? 1 : 0

  depends_on = [huaweicloud_modelarts_devserver_action.test]

  devserver_id = huaweicloud_modelarts_devserver.test.id
  action       = "%[2]s"
}
`, testAccDevServer_basic(sAutoRenew, name, password), actionType, doRetryAction)
}
```
// step 3 to 6 are using it

File: docs/resources/modelarts_devserver_action.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  Changing this creates a new resource.
+
+* `devserver_id` - (Required, String, ForceNew) Specifies the ID of the DevServer.
+  Changing this creates a new resource.
+
+* `action` - (Required, String, ForceNew) Specifies the action type of the DevServer.
+  Changing this creates a new resource.  
+  The valid values are as follows:
+  + **start**: The DevServer can be started only when the DevServers is stopped, fails to stop, or fails to start.
+  + **stop**: The DevServer can be stopped only when it is running or stop failure.
```
Comment (Lance52259) [inline docs/resources/modelarts_devserver_action.md]:
```suggestion
  + **start**: The DevServer can be started only when the DevServers is stopped, stop failure, or start failure.
  + **stop**: The DevServer can be stopped only when it is running or stop failure.
```

**Comment by wuzhuanhong** [comment]
./scripts/coverage.sh -o modelarts -f TestAccDevServer_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/modelarts" -v -coverprofile="./huaweicloud/services/acceptance/modelarts/modelarts_coverage.cov" -coverpkg="./huaweicloud/services/modelarts" -run TestAccDevServer_basic -timeout 360m -parallel 10
=== RUN   TestAccDevServer_basic
=== PAUSE TestAccDevServer_basic
=== CONT  TestAccDevServer_basic
--- PASS: TestAccDevServer_basic (504.73s)
PASS
coverage: 10.1% of statements in ./huaweicloud/services/modelarts
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts 504.817s        coverage: 10.1% of statements in ./huaweicloud/services/modelarts

**Comment by wuzhuanhong** [comment]
./scripts/coverage.sh -o modelarts -f TestAccDevServer_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/modelarts" -v -coverprofile="./huaweicloud/services/acceptance/modelarts/modelarts_coverage.cov" -coverpkg="./huaweicloud/services/modelarts" -run TestAccDevServer_basic -timeout 360m -parallel 10
=== RUN   TestAccDevServer_basic
=== PAUSE TestAccDevServer_basic
=== CONT  TestAccDevServer_basic
--- PASS: TestAccDevServer_basic (446.74s)
PASS
coverage: 10.1% of statements in ./huaweicloud/services/modelarts
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts 446.849s        coverage: 10.1% of statements in ./huaweicloud/services/modelarts

**Pull request merged**